### PR TITLE
feat: superfluid one off payment processor

### PR DESCRIPTION
**Pull request by leoslr**
## Description of the changes

Adds payment processor logic and tests for superfluid one-off payment.
Also adds the intermediary prepare function for escrow approval/payment and superfluid wrapping/unwrapping.

### CODE REVIEW

File: packages/payment-processor/package.json [hunk 1]
```diff
@@ -39,6 +39,7 @@
     "test:watch": "yarn test --watch"
   },
   "dependencies": {
+    "@openzeppelin/contracts": "4.7.3",
```
Comment (leoslr) [inline packages/payment-processor/package.json:42]: The superfluid framework does not implement the erc777 interface. I had to get the abi from here.

File: packages/payment-processor/test/payment/erc777-stream.test.ts [hunk 2]
```diff
@@ -239,4 +241,53 @@ describe('erc777-stream', () => {
...
+        account: wallet.address,
+        providerOrSigner: provider,
+      });
+      const recipientBalanceAfter = await daix.balanceOf({
+        account: paymentAddress,
+        providerOrSigner: provider,
+      });
+
+      expect(BigNumber.from(senderBalanceBefore)).toEqual(
+        BigNumber.from(senderBalanceAfter).add(oneOffPaymentAmount),
```
Comment (yomarion) [inline packages/payment-processor/test/payment/erc777-stream.test.ts:286]: I don't understand, why does it work? Shouldn't it be `.sub(` the amount?

**Review by yomarion — APPROVED** [review]

File: packages/payment-processor/src/payment/erc777-utils.ts [hunk 3]
```diff
@@ -156,6 +156,22 @@ export async function approveUnderlyingToken(
   return signer.sendTransaction(preparedTx);
 }
 
+/**
+ * Prepare the wrap transaction of the speicified amount of underlying token into supertokens
```
Comment (bertux) [inline packages/payment-processor/src/payment/erc777-utils.ts:160]:
```suggestion
 * Prepare the wrap transaction of the specified amount of underlying token into supertoken
```

File: packages/payment-processor/test/payment/erc777-stream.test.ts (hunk 2 above)
Comment (bertux) [reply packages/payment-processor/test/payment/erc777-stream.test.ts:286]: I guess @yomarion means balanceAfter = balanceBefore - sentAmount

**Review by bertux — APPROVED** [review]
just comment typos commented

**Pull request merged**